src/main/scala/it/pagopa/interop/partymanagement/api/impl/PublicApiServiceImpl.scala

```diff
@@ -189,24 +189,47 @@ class PublicApiServiceImpl(
-        val party: Party =
-          if (institutionParty.origin == ipaOrigin)
-            institutionParty
-              .copy(institutionType = institutionUpdate.institutionType.orElse(institutionParty.institutionType))
-          else
-            institutionParty.copy(
-              institutionType = institutionUpdate.institutionType.orElse(institutionParty.institutionType),
-              address = institutionUpdate.address.getOrElse(institutionParty.address),
-              taxCode = institutionUpdate.taxCode.getOrElse(institutionParty.taxCode),
-              description = institutionUpdate.description.getOrElse(institutionParty.description),
-              digitalAddress = institutionUpdate.digitalAddress.getOrElse(institutionParty.digitalAddress)
-            )
+    val institutionPartyUpdate  = updateWithInstitutionUpdate(institutionParty, relationship)
+    val institutionPartyProduct = updateWithInstitutionProductInfo(institutionPartyUpdate, relationship)
+    Option
+      .when(relationship.institutionUpdate.isDefined || relationship.billing.isDefined) {
+        getCommander(institutionPartyProduct.id.toString).ask(ref => UpdateParty(institutionPartyProduct, ref))
+      }
+      .getOrElse(Future.successful(StatusReply.Success[Party](institutionParty)))
+  }
 
-        getCommander(party.id.toString).ask(ref => UpdateParty(party, ref))
+  private def updateWithInstitutionUpdate(
+    institutionParty: InstitutionParty,
+    relationship: PersistedPartyRelationship
+  ): InstitutionParty = {
+    relationship.institutionUpdate.fold(institutionParty) { institutionUpdate =>
+      if (institutionParty.origin == ipaOrigin)
+        institutionParty
+          .copy(institutionType = institutionUpdate.institutionType.orElse(institutionParty.institutionType))
+      else
+        institutionParty.copy(
+          institutionType = institutionUpdate.institutionType.orElse(institutionParty.institutionType),
+          address = institutionUpdate.address.getOrElse(institutionParty.address),
+          taxCode = institutionUpdate.taxCode.getOrElse(institutionParty.taxCode),
+          description = institutionUpdate.description.getOrElse(institutionParty.description),
+          digitalAddress = institutionUpdate.digitalAddress.getOrElse(institutionParty.digitalAddress),
+          zipCode = institutionUpdate.zipCode.getOrElse(institutionParty.zipCode)
+        )
     }
+  }
 
+  private def updateWithInstitutionProductInfo(
+    institutionParty: InstitutionParty,
+    relationship: PersistedPartyRelationship
+  ): InstitutionParty = {
+    relationship.billing.fold(institutionParty) { billing =>
+      val productId          = relationship.product.id
+      val institutionProduct = institutionParty.products
+        .find(_.product == productId)
+        .getOrElse(PersistedInstitutionProduct(product = productId, pricingPlan = None, billing = null))
+        .copy(pricingPlan = relationship.pricingPlan, billing = billing)
```

Comment:
avoid using `null`s, even if it is temporary 🙏 
```suggestion
        .map(_.copy(pricingPlan = relationship.pricingPlan, billing = billing))
        .getOrElse(PersistedInstitutionProduct(product = productId, pricingPlan = relationship.pricingPlan, billing = billing))
```